pkg/cloud/gcp/actuators/util/gcp_tags_labels_test.go

```diff
@@ -0,0 +1,302 @@
+/*
+Copyright The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"reflect"
+	controllerclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	controllerfake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestGetLabelsList(t *testing.T) {
+
+	infraObj := configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: globalInfrastructureName,
+		},
+		Spec: configv1.InfrastructureSpec{
+			PlatformSpec: configv1.PlatformSpec{
+				Type: configv1.GCPPlatformType,
+			},
+		},
+		Status: configv1.InfrastructureStatus{
+			InfrastructureName: "test-3546b",
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.GCPPlatformType,
+				GCP:  &configv1.GCPPlatformStatus{},
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                 string
+		labelsFeatureEnabled bool
+		getInfra             func() *configv1.Infrastructure
+		machineSpecLabels    map[string]string
+		expectedLabels       map[string]string
+		wantErr              bool
+	}{
+		{
+			name:                 "Infrastructure resource doesn't exist",
```

Comment:
This is a new dependency, we should revert this change

Reply:
Comment incorporated. Changed test case to expect Infrastructure to be present when feature is enabled.